internal/crypto/src/time_stamp/http_request.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use async_generic::async_generic;
+use bcder::{decode::Constructed, encode::Values};
+
+use crate::{
+    asn1::rfc3161::{TimeStampReq, TimeStampResp},
+    time_stamp::{
+        response::TimeStampResponse,
+        verify::{verify_time_stamp, verify_time_stamp_async},
+        TimeStampError,
+    },
+};
+
+#[async_generic]
+pub fn default_rfc3161_request(
+    url: &str,
+    headers: Option<Vec<(String, String)>>,
+    data: &[u8],
+    message: &[u8],
+) -> Result<Vec<u8>, TimeStampError> {
+    let request = Constructed::decode(
+        bcder::decode::SliceSource::new(data),
+        bcder::Mode::Der,
+        TimeStampReq::take_from,
+    )
+    .map_err(|_err| TimeStampError::InternalError("failure to decode Constructed TimeStampReq"))?;
+
+    let ts = time_stamp_request_http(url, headers, &request)?;
```

Comment:
Shouldn't `time_stamp_request_http()` be async-generic? As-is, I think `default_rfc3161_request_async()` would block the executor because `time_stamp_request_http()` uses `ureq` which uses blocking I/O.

Reply:
It has to work in the sync path too

Reply:
Won't implement in this PR. Tracking as CAI-7312.